core/typedefs.h

```diff
@@ -440,3 +440,23 @@ inline constexpr bool is_zero_constructible_v = is_zero_constructible<T>::value;
 #define GODOT_MSVC_WARNING_POP
 #define GODOT_MSVC_WARNING_PUSH_AND_IGNORE(m_warning)
 #endif
+
+template <typename T, typename = void>
+struct is_incomplete_type : std::true_type {};
+
+template <typename T>
+struct is_incomplete_type<T, std::void_t<decltype(sizeof(T))>> : std::false_type {};
+
+template <typename T>
+constexpr bool is_incomplete_type_v = is_incomplete_type<T>::value;
+
+#if !defined(SCU_BUILD) && !defined(DEV_ENABLED) && !defined(DEBUG_ENABLED)
```

Comment:
Might actually be more convenient just to disable in DEV and SCU builds, rather than DEBUG. I say that, because most of us make editor builds regularly, and so it enables "quickly" checking the headers locally without relying on CI to find errors, and without having to build export templates.

Reply:
Hrm, I think (hope!) those include guards won't be triggered often for PRs. For this reason, it might be preferable to have them be checked as little as possible during normal development.
Especially, I was thinking people might want to add includes during dev to 'just make something work temporarily' without needing to worry about the include guards. Most times they'll probably use `dev_mode=yes`, but perhaps sometimes they won't.

However, I could really see it going either way. If you feel strongly about this, I'm happy to adjust.